src/Core/src/Graphics/MauiDrawable.Android.cs

```diff
@@ -389,9 +389,13 @@ protected override void OnDraw(Shape? shape, Canvas? canvas, APaint? paint)
 
 					if (_shape != null)
 					{
-						var bounds = new Graphics.Rect(_strokeThickness / 2, _strokeThickness / 2, _width - _strokeThickness, _height - _strokeThickness);
+						var offset = _strokeThickness / 2 / _density;
+						var w = (_width - _strokeThickness) / _density;
+						var h = (_height - _strokeThickness) / _density;
+
+						var bounds = new Graphics.Rect(offset, offset, w, h);
 						var path = _shape.PathForBounds(bounds);
-						var clipPath = path?.AsAndroidPath();
+						var clipPath = path?.AsAndroidPath(scaleX: (float)_density, scaleY: (float)_density);
```

Comment:
Once we have the cross-platform path, we convert it into the platform path - applying the density back as we do it.